From c7792bdefe08c7ffac3db82420bef230fca1ffc7 Mon Sep 17 00:00:00 2001
From: CI Assistant <chdauto@intel.com>
Date: Wed, 5 Feb 2020 20:46:40 +0200
Subject: [PATCH] Merge pull request #1729 in SW_UGW/linux from
 bugfix/UGW_SW-47114-software-queues-are-not-getting-added-841 to 8.4.1

* commit 'f96776ee36c28854e114b64bd195debd1f533421':
  UGW_SW-47114: Skip netdevice ndo_setup_tc for XRX500
---
 drivers/net/datapath/dpm/datapath_instance.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/datapath/dpm/datapath_instance.c b/drivers/net/datapath/dpm/datapath_instance.c
index d553c7f8de43..a8ea68fcae24 100644
--- a/drivers/net/datapath/dpm/datapath_instance.c
+++ b/drivers/net/datapath/dpm/datapath_instance.c
@@ -298,6 +298,7 @@ int dp_inst_add_dev(struct net_device *dev, char *subif_name, int inst,
 		}
 		hlist_add_head(&dp_dev->hlist, &dp_dev_list[idx]);
 #if IS_ENABLED(CONFIG_PPA)
+#if (!IS_ENABLED(CONFIG_SOC_GRX500))
 		/*backup ops*/
 		if (dev) {
 			dev->features |= NETIF_F_HW_TC;
@@ -311,7 +312,8 @@ int dp_inst_add_dev(struct net_device *dev, char *subif_name, int inst,
 			if (err)
 				return DP_FAILURE;
 		}
-#endif
+#endif /* CONFIG_SOC_GRX500 */
+#endif /* CONFIG_PPA */
 #if IS_ENABLED(CONFIG_INTEL_DATAPATH_SWITCHDEV)
 	if (!(flag & DP_F_SUBIF_LOGICAL))
 		dp_port_register_switchdev(dp_dev, dev);
